packages/xo-web/src/xo-app/xosan/new-xosan.js

```diff
@@ -195,10 +241,21 @@ export default class NewXosan extends Component {
     this.setState({ brickSize })
   }
 
-  _selectSr = async (event, sr) => {
-    const selectedSrs = { ...this.state.selectedSrs }
-    selectedSrs[sr.id] = event.target.checked
-    this.setState({ selectedSrs })
+  _selectSrs = selectedSrs => {
+    const found = {}
+    const { srs } = this.props
+    if (
+      some(selectedSrs, srId => {
+        const container = srs[srId].$container
+        const foundContainer = found[container]
+        found[container] = true
+        return foundContainer
+      })
+    ) {
+      return this.setState({ srsOnSameHost: true })
```

Comment:
You still need to update the `selectedSrs`, right?
What about something like this:
```js
_selectSrs = selectedSrs => {
  const { srs } = this.props
  const found = {}
  let container
  this.setState({
    selectedSrs,
    srsOnSameHost: some(selectedSrs, srId => {
      container = get(() => srs[srId].$container)
      return found[container] || (found[container] = true, false)
    }),
  })
}
```

Reply:
Also, it would be better if `_selectSrs` would only `setState` as before and the duplication check was done in a selector instead.

Reply:
We don't need to update `selectedSrs` if  it contain srs on the same host.

Reply:
Why not?